### PR TITLE
Add Dapr runtime version and enabled features to Metadata API

### PR DESCRIPTION
**Pull request by shubham1172**
# Description

Co-authored-by: @ItalyPaleAle 

Add a runtime version to metadata root. It is kept in extended metadata for back-compat but should be deprecated later.
This also adds an 'enabledFeatures' key with list of preview features enabled.

```bash
daprd --app-id test --app-port 3001 --app-protocol grpc --dapr-http-port 3501 -- sleep 1000
curl localhost:3501/v1.0/metadata | jq .
```
```json
{
  "id": "test",
  "runtimeVersion": "edge",
  "enabledFeatures": [
    "ServiceInvocationStreaming"
  ],
  "extended": {
    "daprRuntimeVersion": "edge"
  },
  "appConnectionProperties": {
    "port": 3001,
    "protocol": "grpc",
    "channelAddress": "127.0.0.1"
  },
}
```

## Issue reference

Please reference the issue this PR will close: #6528

## Checklist

Please make sure you've  completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [x] Created/updated tests
* [x] Unit tests passing
* [x] End-to-end tests passing
* [x] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: https://github.com/dapr/docs/issues/3557
* [ ] Specification has been updated / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Provided sample for the feature / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

File: pkg/grpc/api_test.go [hunk 1]
```diff
@@ -4341,7 +4341,8 @@ func TestMetadata(t *testing.T) {
 			`"subscriptions":[{"pubsub_name":"test","topic":"topic","rules":{"rules":[{"path":"path"}]},"dead_letter_topic":"dead"}],` +
 			`"http_endpoints":[{"name":"MockHTTPEndpoint"}],` +
 			`"app_connection_properties":{"port":5000,"protocol":"grpc","channel_address":"1.2.3.4","max_concurrency":10,` +
-			`"health":{"health_probe_interval":"10s","health_probe_timeout":"5s","health_threshold":3}}}`
+			`"health":{"health_probe_interval":"10s","health_probe_timeout":"5s","health_threshold":3}},` +
```
Comment (ItalyPaleAle) [inline pkg/grpc/api_test.go:4344]: I didn't notice this earlier, but can you please make sure that all keys use camelCase? `healthProbeInterval` etc

**Review by ItalyPaleAle — CHANGES_REQUESTED** [review]

File: pkg/grpc/api_test.go (hunk 1 above)
Comment (shubham1172) [reply pkg/grpc/api_test.go:4344]: This is gRPC, right? Only a part of marshaling in test. In HTTP it's still `camelCase`.

Comment (ItalyPaleAle) [reply pkg/grpc/api_test.go:4344]: Right!

**Review by ItalyPaleAle — DISMISSED** [review]

**Review by daixiang0 — DISMISSED** [review]
LGTM

**Review by ItalyPaleAle — APPROVED** [review]

**Review by daixiang0 — APPROVED** [review]

**Comment by shubham1172** [comment]
Converted to draft, see: https://github.com/dapr/docs/pull/3562#discussion_r1233570043

**Review by ItalyPaleAle — APPROVED** [review]

**Review by ItalyPaleAle — APPROVED** [review]

**Review by yaron2 — APPROVED** [review]

**Pull request merged**